src/Domain/POV/ConfigCreator.php

```diff
@@ -0,0 +1,554 @@
+      WHERE l.layer_active = 1 AND p.plan_id IS NULL OR (p.plan_state IN ('IMPLEMENTED') AND p.plan_active = 1) AND
+      (
+        l.layer_geotype IN ('line','point','polygon') OR (
+          # if raster, return only if the region overlaps with its bounding box data
+          -- l.layer_geotype IN ('raster')
+          JSON_EXTRACT(l.layer_raster, '$.boundingbox[0][0]') < :bottomRightX
+          AND JSON_EXTRACT(l.layer_raster, '$.boundingbox[0][1]') < :bottomRightY
+          AND JSON_EXTRACT(l.layer_raster, '$.boundingbox[1][0]') > :topLeftX
+          AND JSON_EXTRACT(l.layer_raster, '$.boundingbox[1][1]') > :topLeftY
+        )
+      )
+      GROUP BY l.layer_id
+  ),
+  # include kpi value if layer is an ecology layer and
+  #   extract new data columns from layer_type json data
+  LayerStep2 AS (
+      SELECT
+        l.*,
+        k.kpi_value,
+        JSON_ARRAYAGG(JSON_OBJECT(
+          'channel', 'r',
+          'max', t.value,
+          'type', t.id-1
+        )) AS layer_type_mapping,
+        JSON_ARRAYAGG(JSON_OBJECT(
+          'name', t.display_name,
+          '//material', 'todo'
+        )) AS layer_type_types,
+        MIN(t.value) AS layer_type_value_min,
+        MAX(t.value) AS layer_type_value_max,
+        COUNT(t.id) as layer_type_value_count,
+        CAST((MAX(t.value) / (COUNT(t.id)-1)) as INT) as layer_type_value_step
+      FROM LayerStep1 l
+      INNER JOIN JSON_TABLE(
+        JSON_EXTRACT(l.layer_type, '$.*'),
+          '$[*]' COLUMNS (
+            id for ordinality,
+            display_name VARCHAR(255) PATH '$.displayName',
+            value int PATH '$.value'
+        )
+      ) AS t
+      LEFT JOIN LatestEcologyKpiFinal k ON (
+        l.layer_short != '' AND CONCAT('mel_',LOWER(REPLACE(k.kpi_name,' ','_'))) = l.layer_name
+      )
+      GROUP BY l.layer_id
+  ),
+  LayerFinal AS (
+      SELECT
+        l.*,
+        JSON_ARRAYAGG(
+          JSON_OBJECT(
+            'points', JSON_EXTRACT(g.geometry_geometry, '$'),
+            'types', JSON_ARRAY(JSON_EXTRACT(g.geometry_type, '$')),
+            'gaps', IF(g.geometry_gaps=JSON_ARRAY(null),JSON_ARRAY(), g.geometry_gaps),
+            # just add some aliases to the metadata
+            'metadata', JSON_MERGE_PATCH(
+              g.geometry_data,
+              JSON_OBJECT(
+                'name', JSON_EXTRACT(g.geometry_data, '$.Name'),
+                'turbines', JSON_EXTRACT(g.geometry_data, '$.N_TURBINES'),   
+                '//width', 'todo, eg: 1',
+                '//direction', 'todo, eg: true',
+                '//number_cables', 'todo, eg: 1'
+              )
+            )
+          )        
+        ) AS layer_data
+      FROM LayerStep2 l
+      # left join since raster layers do not ever have geometry data
+      LEFT JOIN LatestGeometryInRegion as g ON g.geometry_layer_id=l.layer_id
+      WHERE l.layer_geotype IN ('raster') OR g.geometry_layer_id IS NOT NULL
+      GROUP BY l.layer_id
+  )
```

Comment:
so the WITH part are all prepared temporary table to be used by the actual main SELECT below

Reply:
wow!